File: server/src/ChinaImportPlatform.Api/Models/Message.cs
namespace ChinaImportPlatform.Api.Models;

/// <summary>A chat message. Stored server-side so history survives device changes.</summary>
public class Message : BaseEntity
{
    public Guid ConversationId { get; set; }

    public Guid SenderId { get; set; }

    /// <summary>Optionally attaches the message to a specific order for context.</summary>
    public Guid? OrderId { get; set; }

    public string Body { get; set; } = string.Empty;

    public string? AttachmentS3Key { get; set; }

    public DateTime SentAt { get; set; } = DateTime.UtcNow;

    public DateTime? DeliveredAt { get; set; }

    public DateTime? ReadAt { get; set; }
}

File: client/china_import_app/lib/features/messaging/domain/entities/message.dart
/// Domain representation of a single chat message.
class Message {
  const Message({
    required this.id,
    required this.conversationId,
    required this.body,
    required this.isMine,
    required this.sentAt,
  });

  final String id;
  final String conversationId;
  final String body;
  final bool isMine;
  final DateTime sentAt;
}
